### PR TITLE
Karpenter Subnet Selector Regx to support AWS Accounts Subnet Tag values

### PR DESCRIPTION
**Pull request by gowripadmajas**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -156,6 +156,18 @@ variable "subnet_ids" {
...
 
+variable "subnet_selector_regx" {
+  description = "Select the Subnets from this Regular expression to spin up the Karpenter nodes"
+  type        = string
+  default     = "aft-app-ap-southeast*"
+
+  validation {
+    condition     = can(regex("^aft-app-ap-southeast*|^App*|^Web*", var.subnet_selector_regx))
+    error_message = "Err: invalid subnet selector, must be one of aft-app-ap-southeast*|App*|Web*."
+  }
```
Comment (niroz89) [inline variables.tf:167]: make it mandatory instead, since its public module.. better not to use these internal naming convention.

Comment (gowripadmajas) [reply variables.tf:167]: my bad.. forgot this is public module :)

File: karpenter.tf [hunk 2]
```diff
@@ -9,7 +9,7 @@ locals {
       nodeclass_name = "default"
       karpenter_subnet_selector_maps = [{
         tags = {
-          "Name" = "aft-app-ap-southeast*"
+          "Name" = var.subnet_selector_regx
```
Comment (niroz89) [inline karpenter.tf:12]:
```suggestion
          "Name" = var.subnet_selector_name_tag
```

File: variables.tf [hunk 3]
```diff
@@ -156,6 +156,11 @@ variable "subnet_ids" {
   type        = list(string)
 }
 
+variable "subnet_selector_regx" {
```
Comment (niroz89) [inline variables.tf:159]:
```suggestion
variable "subnet_selector_name_tag" {
```

File: variables.tf [hunk 4]
```diff
@@ -156,6 +156,11 @@ variable "subnet_ids" {
...
+  description = "Select the Subnets from this Regular expression to spin up the Karpenter nodes"
```
Comment (niroz89) [inline variables.tf:160]:
```suggestion
  description = "Subnet selector Name tags for Karpenter nodes, supports wildcard."
```

**Review by uchinda-sph — APPROVED** [review]

**Pull request merged**